client/reader/stream/index.jsx

```diff
@@ -214,48 +163,31 @@ class ReaderStream extends React.Component {
 	}
 
 	componentWillReceiveProps( nextProps ) {
-		if ( nextProps.postsStore !== this.props.postsStore ) {
-			this.props.postsStore.off( 'change', this.updateState );
-			this.props.recommendationsStore &&
-				this.props.recommendationsStore.off( 'change', this.updateState );
-
-			nextProps.postsStore.on( 'change', this.updateState );
-			nextProps.recommendationsStore &&
-				nextProps.recommendationsStore.on( 'change', this.updateState );
+		const { streamKey } = nextProps;
+		if ( streamKey !== this.props.streamKey ) {
 			this.props.resetCardExpansions();
-
-			this.updateState( nextProps );
-			this._list && this._list.reset();
+			this.props.viewStream( { streamKey } );
```

Comment:
damn, you just made me realize that `page` doesn't belong anywhere in React per se.  It isn't tied to rerenders, or the mounting/dismounting of components.  It shouldn't be reset when `streamKey` is changed. There needs to be a separate page held on to by each streamKey that is incremented (and remembered).

I'll move this logic to the `analyticsAlgoMap` that I've already created in the data-layer, which is a better fit anyway.

Reply:
I started refactoring all of the analytics that occur in `Streams` to isolate where they live instead of existing partially in a few places (data-layer, stream component, and controller).  In particular it would be ideal if the config for analytics per stream lived in just one place.

As I was making the sweeping changes I realized that it was going to be much too large to review on top of this already massive PR.  I've started the work in a different branch, and for the purposes of this PR I think we can be best served by maintaining page counts in a simple map at the module level of the `Stream` react component.